scene/resources/text_file.cpp

```diff
@@ -0,0 +1,147 @@
+/* distribute, sublicense, and/or sell copies of the Software, and to    */
+/* permit persons to whom the Software is furnished to do so, subject to */
+/* the following conditions:                                             */
+/*                                                                       */
+/* The above copyright notice and this permission notice shall be        */
+/* included in all copies or substantial portions of the Software.       */
+/*                                                                       */
+/* THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND,       */
+/* EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF    */
+/* MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT.*/
+/* IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY  */
+/* CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION OF CONTRACT,  */
+/* TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION WITH THE     */
+/* SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.                */
+/*************************************************************************/
+
+#include "text_file.h"
+
+#include "os/file_access.h"
+
+bool TextFile::has_text() const {
+	return text != "";
+}
+
+String TextFile::get_text() const {
+	return text;
+}
+
+void TextFile::set_text(const String &p_code) {
+	text = p_code;
+}
+
+Error TextFile::load_text(const String &p_path) {
+
+	PoolVector<uint8_t> sourcef;
+	Error err;
+	FileAccess *f = FileAccess::open(p_path, FileAccess::READ, &err);
+	if (err) {
+		ERR_FAIL_COND_V(err, err);
+	}
+
+	int len = f->get_len();
+	sourcef.resize(len + 1);
+	PoolVector<uint8_t>::Write w = sourcef.write();
+	int r = f->get_buffer(w.ptr(), len);
+	f->close();
+	memdelete(f);
+	ERR_FAIL_COND_V(r != len, ERR_CANT_OPEN);
+	w[len] = 0;
+
+	String s;
+	if (s.parse_utf8((const char *)w.ptr())) {
+		ERR_EXPLAIN("Script '" + p_path + "' contains invalid unicode (utf-8), so it was not loaded. Please ensure that scripts are saved in valid utf-8 unicode.");
+		ERR_FAIL_V(ERR_INVALID_DATA);
+	}
+	text = s;
+	path = p_path;
+	return OK;
+}
+
+////////////////////////////////////////////////////////////////////////////////
+
+RES ResourceFormatLoaderTextFile::load(const String &p_path, const String &p_original_path, Error *r_error) {
+	if (r_error) {
+		*r_error = ERR_FILE_CANT_OPEN;
+	}
+
+	TextFile *text_file = memnew(TextFile);
+
+	Ref<TextFile> text_res(text_file);
+	Error err = text_file->load_text(p_path);
+
+	if (err != OK) {
+		ERR_FAIL_COND_V(err != OK, RES());
+	}
+
+	text_file->set_file_path(p_original_path);
+	text_file->set_path(p_original_path);
+
+	if (r_error) {
+		*r_error = OK;
+	}
+
+	return text_res;
+}
+
+void ResourceFormatLoaderTextFile::get_recognized_extensions(List<String> *p_extensions) const {
+	p_extensions->push_back("*");
+}
+
+bool ResourceFormatLoaderTextFile::handles_type(const String &p_type) const {
+	return (p_type == "TextFile");
```

Comment:
I still did not completely get the idea of `handles_type` Is passed when the editor is searching for a ui element to handle a specific resource? so I could edit a tscn with the type TextFile and with the type Scene and based on that the editor would behave differently?

I think I got it wrong because than it would make no sense why it is in Resource**Format**LoaderTextFile

Reply:
okay Thank you!
so it is basically `load(file_on_disc, as_resource_type) `(if possible ;) )